CMakeLists.txt

```diff
@@ -33,7 +33,7 @@ endif()
 
 find_package(Threads)
 
-add_library(abieos STATIC src/abi.cpp src/crypto.cpp include/eosio/fpconv.c)
+add_library(abieos STATIC src/abi.cpp src/crypto.cpp include/eosio/fpconv.c src/abieos.cpp)
```

Comment:
The pattern seems to be to only include `src/abieos.cpp` in the executables that link to `abieos` and not in `abieos`. 

There is also `abieos_module` which may be the more appropriate `target_link_library` for `hex_to_json`, was that attempted and failed for some reason?

Reply:
having in tools/CMakeLists
`target_link_libraries(hex_to_json abieos_module ${CMAKE_THREAD_LIBS_INIT})
`
resulted in
```
CMake Error at tools/CMakeLists.txt:12 (target_link_libraries):
  Target "abieos_module" of type MODULE_LIBRARY may not be linked into
  another target.  One may link only to INTERFACE, OBJECT, STATIC or SHARED
  libraries, or to executables with the ENABLE_EXPORTS property set.

```